pyW215/pyW215.py

```diff
@@ -57,6 +68,19 @@ def __init__(self, ip, password, user = "admin",
         if self.use_legacy_protocol:
             _LOGGER.info("Enabled support for legacy firmware.")
         self._error_report = False
+        self.dev_id = DEFAULT_DEV_ID
+        if self.use_legacy_protocol:
+            try:
+                hw_ver = self.fetchMyCgi()['HW Ver']
+                # change dev map for HW A2
+                if hw_ver == 'A2':
+                    self.dev_id = W215_A2_DEV_ID
+                # this assume the model number is W215, might be wrong
+                self.model_name = 'W215_' + hw_ver
+            except:
+                self.model_name = 'Unknown'
+        else:
+            self.model_name = self.SOAPAction("", "ModelName", "")
```

Comment:
Could we change this to

` self.model_name = self.SOAPAction(Action="", responseElement="ModelName", params = "")`

As suggested in the PR from @clach04. It is much more readable.